### PR TITLE
Adding opacity -> alpha method to Color class

### PR DESCRIPTION
**Pull request by Piinks**
This change stems from a suggestion made in https://github.com/flutter/flutter/pull/44289/files#r347058942

This simple method shows up a couple of times in the flutter framework. Adding it to dart:ui seems generally useful. :)

### CODE REVIEW

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -307,6 +307,15 @@ class Color {
     }
   }
 
+  /// Returns an alpha value representative of the provided [opacity] value.
+  ///
+  /// The [opacity] value may not be null.
+  static int getAlphaFromOpacity(double opacity) {
+    assert(opacity != null);
+    final double clampedOpacity = opacity.clamp(0.0, 1.0);
+    return (clampedOpacity * 255).round();
```
Comment (gspencergoog) [inline lib/ui/painting.dart]: You could probably combine this all into one line:
```suggestion
    return (opacity.clamp(0.0, 1.0) * 255).round();
```

File: lib/web_ui/lib/src/ui/painting.dart [hunk 2]
```diff
@@ -226,6 +226,15 @@ class Color {
     }
   }
 
+  /// Returns an alpha value representative of the provided [opacity] value.
+  ///
+  /// The [opacity] value may not be null.
+  static int getAlphaFromOpacity(double opacity) {
+    assert(opacity != null);
+    final double clampedOpacity = opacity.clamp(0.0, 1.0);
+    return (clampedOpacity * 255).round();
```
Comment (gspencergoog) [inline lib/web_ui/lib/src/ui/painting.dart]: Ditto:
```suggestion
    return (opacity.clamp(0.0, 1.0) * 255).round();
```

File: lib/ui/painting.dart [hunk 3]
```diff
@@ -307,6 +307,15 @@ class Color {
     }
   }
 
+  /// Returns an alpha value representative of the provided [opacity] value.
+  ///
+  /// The [opacity] value may not be null.
+  static int getAlphaFromOpacity(double opacity) {
+    assert(opacity != null);
+    final double clampedOpacity = opacity.clamp(0.0, 1.0);
```
Comment (Piinks) [inline lib/ui/painting.dart]:
```suggestion
```

File: lib/web_ui/lib/src/ui/painting.dart [hunk 4]
```diff
@@ -226,6 +226,15 @@ class Color {
     }
   }
 
+  /// Returns an alpha value representative of the provided [opacity] value.
+  ///
+  /// The [opacity] value may not be null.
+  static int getAlphaFromOpacity(double opacity) {
+    assert(opacity != null);
+    final double clampedOpacity = opacity.clamp(0.0, 1.0);
```
Comment (Piinks) [inline lib/web_ui/lib/src/ui/painting.dart]:
```suggestion
```

**Pull request merged**